server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -903,6 +906,24 @@ public String toString() {
         return "IngestDocument{" + " sourceAndMetadata=" + ctxMap + ", ingestMetadata=" + ingestMetadata + '}';
     }
 
+    public void redirect(String destIndex) {
+        getMetadata().setIndex(destIndex);
+        invokeDefaultPipelineOfDestination = true;
+        skipCurrentPipeline = true;
+    }
+
+    boolean isInvokeDefaultPipelineOfDestination() {
+        return invokeDefaultPipelineOfDestination;
+    }
+
+    boolean isSkipCurrentPipeline() {
+        return skipCurrentPipeline;
+    }
```

Comment:
I don't think we need multiple boolean flags for this right? If we do, then it's not clear enough why they have to be separated (needs documentation)

Reply:
I've changed it to just be a single boolean and a single getter for it (`isRedirect()`). The downside is that `isRedirect()` is less expressive as `isSkipCurrentPipeline()` in the context of determining whether to skip the pipeline.
We could keep `isSkipCurrentPipeline()` and just return the `redirect` flag. But that feels a bit off as well.